pandas/tests/test_util.py

```diff
@@ -76,6 +78,59 @@ def test_rands_array():
     assert(len(arr[1, 1]) == 7)
 
 
+class TestToNumeric(tm.TestCase):
+    def test_series(self):
+        s = pd.Series(['1', '-3.14', '7'])
+        res = to_numeric(s)
+        expected = pd.Series([1, -3.14, 7])
+        assert_series_equal(res, expected)
+
+        s = pd.Series(['1', '-3.14', 7])
+        res = to_numeric(s)
+        assert_series_equal(res, expected)
+
+    def test_no_box(self):
+        s = pd.Series(['1', '-3.14', '7'])
+        res = to_numeric(s, box=False)
+        expected = np.array([1, -3.14, 7])
+        assert_numpy_array_equal(res, expected)
+
+        s = pd.Series([1, -3.14, 'apple'])
+        res = to_numeric(np.array(s), box=False, errors='ignore')
+        expected = np.array([1, -3.14, 'apple'], dtype='O')
+        for x,y in zip(res, expected):
+            assert_true(x == y)
+        assert_true(isinstance(res, np.ndarray))
+
+    def test_error(self):
+        s = pd.Series([1, -3.14, 'apple'])
+        assert_raises(ValueError, to_numeric, s, errors='raise')
+
+        res = to_numeric(s, errors='ignore')
+        expected = pd.Series([1, -3.14, 'apple'])
+        assert_series_equal(res, expected)
+
+        res = to_numeric(s, errors='coerce')
+        expected = pd.Series([1, -3.14, np.nan])
+        assert_series_equal(res, expected)
```

Comment:
Can you add a test here with all NaNs? (as this was a behaviour in convert_objects that was changed)

eg

```
In [4]: s = pd.Series(['pear', 'orange', 'apple'])

In [5]: pd.to_numeric(s, errors='coerce')
Out[5]:
0   NaN
1   NaN
2   NaN
dtype: float64
```